lib/tasks/hosts.rake

```diff
@@ -33,15 +39,20 @@ namespace :hosts do
     pingable = []
     missingdns = []
     offline = []
+    long_interval = (
+      2 * (Setting[:puppet_interval] + Setting[:outofsync_interval])
+    ).seconds.ago
 
-    Host.out_of_sync(1.hour.ago).all(:order => 'environment_id asc').collect do |host|
+    Host::Managed.out_of_sync(
+      ENV['OUTOFSYNC_INTERVAL'].to_i.seconds.ago || long_interval
+    ).order('environment_id ASC').collect do |host|
       $stdout.flush
       ip = Resolv::DNS.new.getaddress(host.name).to_s rescue nil
       if ip.empty?
         missingdns << host
       else
         puts "conflict IP address for #{host.name}" unless ip == host.ip
```

Comment:
we can just check that the ip is part of one the interfaces we have assigned to the host.
e.g.``` host.all_ips.include?(ip)```